### PR TITLE
GHA: Use python-pillow/Pillow's Codecov token for own builds

### PR DESCRIPTION
**Pull request by hugovk**
Currently we're using the Codecov token of python-pillow/Pillow for all builds, including builds in forks. This means we get coverage reports at Codecov from those builds, and fork builds do not get coverage for their own builds.

The Codecov tool can automatically figure out the repo, and a token isn't needed, on Travis CI and AppVeyor, but not yet for GitHub Actions.

So, let's remove the token from the config file.

* Then forks will get their own coverage for Travis and AppVeyor (but not GHA).

* No fork coverage is sent upstream.

But if we're doing a python-pillow/Pillow build on GHA, then first copy a config file containing the token into the root, so we do get GHA coverage.

Once Codecov update to autodetect on GHA, we can remove this workaround.

---

If you do want GHA coverage from your fork builds, fetch your token from https://codecov.io/gh/my-username/Pillow/settings and add it as `CODECOV_TOKEN` at https://github.com/my-username/Pillow/settings/secrets.

We cannot rely on this secret alone for upstream, because secrets are not available to PRs, and we especially want coverage for PRs.

### CODE REVIEW

**Comment by hugovk** [comment]
Example build from my fork with a token in secrets:

* Coverage for Travis, AppVeyor, GHA

* https://codecov.io/gh/hugovk/Pillow/commit/f7221be2fc50a8217bc395331375eca782d3befe/build

Example build from my fork with no token in secrets:

* Coverage for Travis, AppVeyor (no GHA)

* https://codecov.io/gh/hugovk/Pillow/commit/4a105a82144f98f3dba9d066384ef80dc319f0b3

Neither of these commits show up on CodeCov for upstream:

* https://codecov.io/gh/python-pillow/Pillow/commits

**Comment by hugovk** [comment]
Hmm, some unrelated GHA error, also happening with master right now. Will restart this later.

```
et:53 http://ppa.launchpad.net/ondrej/php/ubuntu bionic/main amd64 Packages [51.8 kB]
Reading package lists...
W: Conflicting distribution: https://packages.microsoft.com/repos/microsoft-ubuntu-bionic-prod bionic InRelease (expected bionic but got stable)
E: Repository 'https://packages.microsoft.com/repos/microsoft-ubuntu-bionic-prod bionic InRelease' changed its 'Origin' value from 'microsoft-ubuntu-bionic-prod bionic' to 'vscode stable'
E: Repository 'https://packages.microsoft.com/repos/microsoft-ubuntu-bionic-prod bionic InRelease' changed its 'Label' value from 'microsoft-ubuntu-bionic-prod bionic' to 'vscode stable'
E: Repository 'https://packages.microsoft.com/repos/microsoft-ubuntu-bionic-prod bionic InRelease' changed its 'Suite' value from 'bionic' to 'stable'
E: Repository 'https://packages.microsoft.com/repos/microsoft-ubuntu-bionic-prod bionic InRelease' changed its 'Codename' value from 'bionic' to 'stable'
W: Conflicting distribution: https://packages.microsoft.com/ubuntu/16.04/prod xenial InRelease (expected xenial but got stable)
E: Repository 'https://packages.microsoft.com/ubuntu/16.04/prod xenial InRelease' changed its 'Origin' value from 'microsoft-ubuntu-xenial-prod xenial' to 'vscode stable'
E: Repository 'https://packages.microsoft.com/ubuntu/16.04/prod xenial InRelease' changed its 'Label' value from 'microsoft-ubuntu-xenial-prod xenial' to 'vscode stable'
E: Repository 'https://packages.microsoft.com/ubuntu/16.04/prod xenial InRelease' changed its 'Suite' value from 'xenial' to 'stable'
E: Repository 'https://packages.microsoft.com/ubuntu/16.04/prod xenial InRelease' changed its 'Codename' value from 'xenial' to 'stable'
##[error]Process completed with exit code 100.
```

**Comment by hugovk** [comment]
All green now!

File: .github/workflows/codecov-upstream.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Documentation: https://docs.codecov.io/docs/codecov-yaml
+
+codecov:
+  # Avoid "Missing base report" due to committing CHANGES.rst with "[CI skip]"
+  # https://github.com/codecov/support/issues/363
+  # https://docs.codecov.io/v4.3.6/docs/comparing-commits
```
Comment (radarhere) [inline .github/workflows/codecov-upstream.yml]:
```suggestion
  # https://docs.codecov.io/docs/comparing-commits
```
Fixes the 404 as per #4267

Comment (radarhere) [inline .github/workflows/codecov-upstream.yml]:
```suggestion
  # https://docs.codecov.io/docs/comparing-commits
```

**Comment by radarhere** [comment]
.github/workflows/codecov-upstream.yml is actually being detected by GHA as a separate job on individual repos.

Screen Shot 2019-12-24 at 12 40 35 pm

**Comment by hugovk** [comment]
Thanks, I've moved it up a level:

image

**Pull request merged**